### PR TITLE
Use more sensible exception class for `detect_old_contact_file()`

### PR DESCRIPTION
**Pull request by MetRonnie**
This is a small change with no associated issue.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes 
- [x] Tests are updated
- [x] Changelog not needed (pretty insignificant to users)
- [x] No docs update
- [x] Refactor on master branch only

### CODE REVIEW

File: cylc/flow/network/client.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def timeout_handler(self) -> None:
         # behind a contact file?
         try:
             detect_old_contact_file(self.workflow)
-        except (AssertionError, ServiceFileError):
+        except ContactFileExists:
```
Comment (hjoliver) [inline cylc/flow/network/client.py:150]: Any idea why `AssertionError` was there?

File: cylc/flow/clean.py [hunk 2]
```diff
@@ -124,7 +125,7 @@ def _clean_check(opts: 'Values', reg: str, run_dir: Path) -> None:
         return
     try:
         detect_old_contact_file(reg)
-    except ServiceFileError as exc:
+    except ContactFileExists as exc:
```
Comment (hjoliver) [inline cylc/flow/clean.py:128]: Not sure what we were thinking, using ServiceFileError in this context!

**Review by hjoliver — APPROVED** [review]

File: cylc/flow/network/client.py (hunk 1 above)
Comment (MetRonnie) [reply cylc/flow/network/client.py:150]: Did a quick git blame time travel, there used to be an `AssertionError` raised in `detect_old_contact_file()` but no more

File: cylc/flow/workflow_files.py [hunk 3]
```diff
@@ -456,7 +457,7 @@ def detect_old_contact_file(
         CylcError:
             If it is not possible to tell for sure if the workflow is running
             or not.
-        ServiceFileError(CylcError):
+        ContactFileExists:
             If old contact file exists and the workflow process still alive.
```
Comment (oliver-sanders) [inline cylc/flow/workflow_files.py:462]:
```suggestion
            If old contact file exists and the workflow process still alive.
        ServiceFileError:
            For corrupt / incompatible contact files.

```

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**